### ISSUE TITLE
Unable to control `Cannot create consistent method ordering`

### ISSUE THREAD

**Issue by multimeric** [microsoft/pyright#10338]
**Describe the bug**
There is no way to disable the `Cannot create consistent method ordering`, and it seemingly has no `report*` option.

**Code or Screenshots**
Suppose we have the following class hierarchy:
```python
class A:
    pass

class B(A):
    pass

class Meta(type):
    def __new__(mcls, name, bases, namespace, **kwargs):
        return super().__new__(mcls, name, bases[0:1], namespace, **kwargs)

class C(A, B, metaclass=Meta):
    pass
```

This works at runtime because we have resolved the mro issue via the metaclass.

However, `pyright` says
```
test.py:11:7 - error: Cannot create consistent method ordering
1 error, 0 warnings, 0 informations
```
There is also no error code in brackets, so it's not clear what error class it is.

**VS Code extension or command-line**
```bash
$ pyright --version
pyright 1.1.352
$ python --version
Python 3.12.6
```

**Comment by erictraut** [comment]
The error class is `reportGeneralTypeIssues`. Since you're performing "magic" in the metaclass that a static type checker cannot understand, you can suppress this diagnostic using a `# type: ignore` or a `# pyright: ignore[reportGeneralTypeIssues]` comment.

**Comment by erictraut** [comment]
You're using a version of pyright that is very outdated. Please update to the latest version (1.1.399).

**Comment by erictraut** [comment]
The suppression comment should be added to the line where the diagnostic is reported. See [this documentation](https://microsoft.github.io/pyright/#/comments?id=line-level-diagnostic-suppression) for details.
